test/modules/standard/gmp/ferguson/gmp_dist_array.chpl

```diff
@@ -44,16 +41,27 @@ modulus.nextprime(modulus);
 writeln(modulus);
 sum.mod(sum, modulus);
 writeln(sum);
-delete modulus;
-delete sum;
+
 
 // Release the array elements.
+// TODO: remove this section once the records are being released correctly.
```

Comment:
I chose to ‘fix’ the leak because:

1)       It still leaks other things, and the information from memleaks correctly guides a developer to the cause of the leak.

a.       I am reasonably sure that when the elements are not leaked, the records will destroy themselves properly.

b.       This will then cause the test to fail (trying to double-free the mpz) which will be loud and noticeable.

2)       The change in scale of what was leaking (on the order of millions of bytes instead of thousands) looked significant.

3)       I could leave a comment explaining that the cleanup is temporary and specifically for this test

From: Michael Noakes <notifications@github.com>
Reply-To: chapel-lang/chapel <reply@reply.github.com>
Date: Thursday, July 28, 2016 at 3:45 PM
To: chapel-lang/chapel <chapel@noreply.github.com>
Cc: Laura Delaney <ldelaney@cray.com>, Author <author@noreply.github.com>
Subject: Re: [chapel-lang/chapel] Convert GMP BigInt class to record (#4243)


In test/modules/standard/gmp/ferguson/gmp_dist_array.chpl<https://github.com/chapel-lang/chapel/pull/4243#discussion_r72716812>:

>

>  // Release the array elements.

> +// TODO: remove this section once the records are being released correctly.
Oh right. That was dumb on my part. Tough call. I believe I would choose to let this test leak then but I can understand someone taking the opposing point of view. In the original version of the app, the app was responsible for memory management. In the new version it’s not supposed to be. We have other leaks in other apps right now that are are due to lower-level bugs. In every case that I know about we’ve elected to leave the leak in the App. I know of one sort-of exception to this rule and was a leak in a popular library that used inherited a leak. We chose to modify the library to cover the leak until we can fix the leak. Mike On Jul 28, 2016, at 2:46 PM, Laura Delaney <notifications@github.com<mailto:notifications@github.com>> wrote: In test/modules/standard/gmp/ferguson/gmp_dist_array.chpl<#4243 (comment)<https://github.com/chapel-lang/chapel/pull/4243#discussion_r72709196>>:
// Release the array elements. +// TODO: remove this section once the records are being released correctly.
I'm not adding manual_free to the BigInt type module, just to that one test so that it leaks less (it still leaks 2000 bytes of other data related to leaking the array elements). If we would prefer to let the test leak everything, we could do that. — You are receiving this because you commented. Reply to this email directly, view it on GitHub<https://github.com/chapel-lang/chapel/pull/4243/files/3bd5d074a99d2427218177f58af6019dc3d8c2be#r72709196>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACERpltxpu50b_M0W-Oi9NEoiKjto-9lks5qaSNTgaJpZM4JTwJ0>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/chapel-lang/chapel/pull/4243/files/3bd5d074a99d2427218177f58af6019dc3d8c2be#r72716812>, or mute the thread<https://github.com/notifications/unsubscribe-auth/APJKMedLul72G3DtVLvVBODO9KJftKDtks5qaTEagaJpZM4JTwJ0>.